tests/testthat/test-survival-workflow-set-fit_best.R

```diff
@@ -0,0 +1,223 @@
+      resamples = sim_rs,
+      grid = grid,
+      metrics = stc_mtrc,
+      control = gctrl
+    )
+
+  expect_silent(static_res <- fit_best(grid_static_res))
+  expect_s3_class(static_res, "workflow")
+  expect_true(is_trained_workflow(static_res))
+})
+
+test_that("grid tuning survival models with integrated metric", {
+  skip_if_not_installed("prodlim")
+
+  # standard setup start -------------------------------------------------------
+
+  set.seed(1)
+  sim_dat <- prodlim::SimSurv(500) %>%
+    mutate(event_time = Surv(time, event)) %>%
+    select(event_time, X1, X2)
+
+  set.seed(2)
+  split <- initial_split(sim_dat)
+  sim_tr <- training(split)
+  sim_te <- testing(split)
+  sim_rs <- vfold_cv(sim_tr)
+
+  time_points <- c(10, 1, 5, 15)
+
+  mod_spec <-
+    proportional_hazards(penalty = tune(), mixture = 1) %>%
+    set_engine("glmnet") %>%
+    set_mode("censored regression")
+
+  grid <- tibble(penalty = 10^c(-4, -2, -1))
+
+  gctrl <- control_grid(save_workflow = TRUE)
+
+  wf_set <-
+    workflow_set(
+      preproc = list(f1 = event_time ~ X1 + X2),
+      models = list(mod = mod_spec)
+    )
+
+  # standard setup end -------------------------------------------------------
+
+  sint_mtrc <- metric_set(brier_survival_integrated)
+
+  set.seed(2193)
+  grid_integrated_res <-
+    workflow_map(
+      wf_set,
+      resamples = sim_rs,
+      grid = grid,
+      metrics = sint_mtrc,
+      eval_time = time_points,
+      control = gctrl
+    )
+
+  expect_silent(integrated_res <- fit_best(grid_integrated_res))
+  expect_s3_class(integrated_res, "workflow")
+  expect_true(is_trained_workflow(integrated_res))
+
+})
+
+test_that("grid tuning survival models with dynamic metric", {
+  skip_if_not_installed("prodlim")
+
+  # standard setup start -------------------------------------------------------
+
+  set.seed(1)
+  sim_dat <- prodlim::SimSurv(500) %>%
+    mutate(event_time = Surv(time, event)) %>%
+    select(event_time, X1, X2)
+
+  set.seed(2)
+  split <- initial_split(sim_dat)
+  sim_tr <- training(split)
+  sim_te <- testing(split)
+  sim_rs <- vfold_cv(sim_tr)
+
+  time_points <- c(10, 1, 5, 15)
+
+  mod_spec <-
+    proportional_hazards(penalty = tune(), mixture = 1) %>%
+    set_engine("glmnet") %>%
+    set_mode("censored regression")
+
+  grid <- tibble(penalty = 10^c(-4, -2, -1))
+
+  gctrl <- control_grid(save_workflow = TRUE)
+
+  wf_set <-
+    workflow_set(
+      preproc = list(f1 = event_time ~ X1 + X2),
+      models = list(mod = mod_spec)
+    )
+
+  # standard setup end -------------------------------------------------------
+
+  dyn_mtrc  <- metric_set(brier_survival)
+
+  set.seed(2193)
+  grid_dynamic_res <-
+    workflow_map(
+      wf_set,
+      resamples = sim_rs,
+      grid = grid,
+      metrics = dyn_mtrc,
+      eval_time = time_points,
+      control = gctrl
+    )
+
+  expect_silent(dynamic_res <- fit_best(grid_dynamic_res))
+  dynamic_res <- fit_best(grid_dynamic_res, eval_time = 1)
```

Comment:
I added these two lines to mirror the existing `fit_best()` tests but am a bit unsure of the intent.